packages/did/src/utils/validation.ts

```diff
@@ -1,7 +1,7 @@
 import { Methods, Chain } from '../models';
 
 export const ethAddrPattern = '0x[A-Fa-f0-9]{40}';
-export const DIDPattern = `^did:${Methods.Erc1056}:(${ethAddrPattern})$`;
+export const DIDPattern = `^did:${Methods.Erc1056}:([A-Za-z0-9]+:)?(${ethAddrPattern})$`;
```

Comment:
```suggestion
export const DIDPattern = `^did:${Methods.Erc1056}:((${Chain.EWC|Chain.VOLTA}):)?(${ethAddrPattern})$`;
```

Reply:
@jrhender we already have two separate methods [isValidErc1056EWC](https://github.com/energywebfoundation/ew-did-registry/blob/6799fe007c1dc9a1e0b43f973fb516938e545673/packages/did/src/utils/validation.ts#L12) and [isValidErc1056VOLTA](https://github.com/energywebfoundation/ew-did-registry/blob/6799fe007c1dc9a1e0b43f973fb516938e545673/packages/did/src/utils/validation.ts#L16) to validate `did` with specific `chain` name.

Reply:
We can also have two methods, one more generic to catch all Erc1056 methods and one just for EWF Erc1056 and it would be `isValidErc1056EWC || isValidErc1056VOLTA`. But since we need this for PDA, which should be network agnostic and not specific just for EWF project I believe, we shouldn't specify chain name in validator.

Reply:
IMO, we could have one general purpose `did` i.e. `did:ethr:address` pattern validator and two separate pattern validator for `ewc` and `volta`. This way we could limit the usability if `did`s to specific chain. So probably we do more strict validation for `did`s wherever needed.

Reply:
Check out my latest suggestion @jrhender @nichonien @JGiter

Reply:
My opinion is that both `did:ethr:addr`, `did:ethr:chainId:addr` are valid DID according to DID spec. If some application doesn't support some chain, then it is responsibility of application to check chain-id-specifier with `parseDID`

Reply:
@JGiter  Is `parseDID` a function we have in `ew-did-registry`?

Reply:
@jrhender We have functions in did/utils/parser which return separately chain and method. We can replace them with `parseDID`